tiledb/sm/query/query_condition.cc

```diff
@@ -86,8 +90,15 @@ Status QueryCondition::init(
     return Status_QueryConditionError("Cannot reinitialize query condition");
   }
 
+  std::string field_name_copy = field_name;
```

Comment:
Why both a `field_name` and a `field_name_ref`?

Reply:
So we can construct both the clause and the AST for now. Should be deleted by next iteration.